src/train_cmd.cpp

```diff
@@ -3785,10 +3788,13 @@ static bool TrainLocoHandler(Train *v, bool mode)
 		CheckNextTrainTile(v);
 	}
 
-	/* Handle stuck trains. */
+	/* Handle train waiting at a PBS signal, where no reservation could be made right now */
 	if (!mode && HasBit(v->flags, VRF_TRAIN_STUCK)) {
 		++v->wait_counter;
 
+		/* Apply waiting penalty once in a while */
+		if (v->wait_counter % 32 == 0) v->AddCargoPotentialPenalty(32);
```

Comment:
this one is 32, whereas previous wait_counters are 64...
named constant please :)

Reply:
And in this kind of waiting `wait_counter` only increments at a rate of 1 per tick.
I'll make a constant and come more thorough comments.